jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessors.java

```diff
@@ -38,6 +44,15 @@
   public static JarProcessor from(
       Path jarPath, TempDirectoryProvider temporaryDirectoryProvider, ProcessingMode mode)
       throws IOException {
+    Integer jarJavaVersion = getJavaMajorVersion(jarPath);
+    if (jarJavaVersion > 11) {
+      throw new IllegalStateException(
+          "This application is in java "
+              + jarJavaVersion
+              + ". It is incompatible with the default base image which only supports versions up to java 11. "
+              + "Please consider specifying a base image of your choice.");
```

Comment:
How about this?

The input JAR (jarPath) is compiled with Java x, but the default base image only supports versions up to Java 11. Specify a custom base image with `--base-image` (if this is the right option).